DocuSignSamplePlugin/DocuSignPlugin/src/main/java/com/ibm/icn/extension/docusign/service/SignRequestService.java

```diff
@@ -105,6 +105,8 @@ public void execute(PluginServiceCallbacks callbacks, HttpServletRequest request
 				{
 					// provide error information
 					callbacks.getLogger().logError(this, methodName, request, e);
+				} finally {
+					UserContext.get().popSubject();
```

Comment:
Subject should only be popped when the corresponding push was successful. If callbacks.getP8Subject fails for example, you'd be popping another subject. If you are not certain that the pushSubject call will be reached and succeed, you may want to add a flag subjectPushed = true after the pushSubject call and check if it's true before popping the subject.